The librt functionality is built into libc on Android.

The Termux git package installs git perl files to $PREFIX/share/git-perl
by patching perl/Makefile.PL, so setup that path when building perl
files in this Makefile.

diff -u -r ../git-2.7.4/Makefile ./Makefile
--- ../git-2.7.4/Makefile	2016-03-17 16:47:59.000000000 -0400
+++ ./Makefile	2016-03-24 22:46:20.335193973 -0400
@@ -1455,7 +1455,6 @@
 
 ifdef HAVE_CLOCK_GETTIME
 	BASIC_CFLAGS += -DHAVE_CLOCK_GETTIME
-	EXTLIBS += -lrt
 endif
 
 ifdef HAVE_CLOCK_MONOTONIC
@@ -1789,7 +1788,7 @@
 	sed -e '1{' \
 	    -e '	s|#!.*perl|#!$(PERL_PATH_SQ)|' \
 	    -e '	h' \
-	    -e '	s=.*=use lib (split(/$(pathsep)/, $$ENV{GITPERLLIB} || "'"$$INSTLIBDIR"'"));=' \
+	    -e '	s=.*=use lib (split(/$(pathsep)/, $$ENV{GITPERLLIB} || "'"@TERMUX_PREFIX@/share/git-perl"'"));=' \
 	    -e '	H' \
 	    -e '	x' \
 	    -e '}' \
